### DISCUSSION TITLE
Difficult to debug events when there are many of them

### DISCUSSION THREAD

**Discussion by DarkPark** [home-assistant/frontend#11271] [Other feature requests]
## The request

Having a lot of devices and trying to debug state changes on the page `Developer Tools` -> `Events` I see huge amount of events from all devices. It's very difficult to find the device I'm interested in in this pile.

## The alternatives

I'd like to suggest to add additional filters - not only by event type but by event content, entity_id, friendly name and so on.
It's also would be nice to be able at any moment to clear all the events dumped (new button `Clear`).
Even better - table view for events with separate columns for event type, event data, time, origin, context ... With ability to filter each column of course.

**Comment by zsarnett** [comment]
We should add a search box and filter
